src/engine/Operation.h

```diff
@@ -71,6 +150,9 @@ class Operation {
         // only at innermost failure of a recursive call
         throw ad_semsearch::AbortException("WEIRD_EXCEPTION");
       }
+      timer.stop();
```

Comment:
So am I seeing this right, when an exception is thrown the timer is not stopped?

Reply:
When an exception is thrown the timer will be deleted (as it only exists in the local scope) and the `runtimeInfo` will not be updated (so the time will be 0). My reasoning was, that in the case that something went wrong in this subtree the `runtimeInfo` would not be interesting, as the timing of the error is normally not important. We could also change the time and then probably set a failure flag in the case of exceptions though.